src/main/java/mate/academy/spring/controller/ShoppingCartController.java

```diff
@@ -0,0 +1,46 @@
+package mate.academy.spring.controller;
+
+import mate.academy.spring.mapper.DtoResponseMapper;
+import mate.academy.spring.model.MovieSession;
+import mate.academy.spring.model.ShoppingCart;
+import mate.academy.spring.model.User;
+import mate.academy.spring.model.dto.response.ShoppingCartResponseDto;
+import mate.academy.spring.service.ShoppingCartService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/shopping-carts")
+public class ShoppingCartController {
+    private final ShoppingCartService shoppingCartService;
+    private final DtoResponseMapper<ShoppingCartResponseDto, ShoppingCart> dtoResponseMapper;
+
+    @Autowired
+    public ShoppingCartController(
+            ShoppingCartService shoppingCartService,
+            DtoResponseMapper<ShoppingCartResponseDto, ShoppingCart> dtoResponseMapper) {
+        this.shoppingCartService = shoppingCartService;
+        this.dtoResponseMapper = dtoResponseMapper;
+    }
+
+    @PutMapping("/movie-sessions")
+    public void updateShoppingCart(@RequestParam Long userId, @RequestParam Long movieSessionId) {
+        User user = new User();
+        user.setId(userId);
+        MovieSession movieSession = new MovieSession();
+        movieSession.setId(movieSessionId);
+        shoppingCartService.addSession(movieSession, user);
```

Comment:
you can simplify it 
```
shoppingCartService.addSession(movieSessionService.get(movieSessionId),
                userService.get(userId));
```

Reply:
I think that it's bad for performance to open additional db connections